File: data-binding/app/src/main/java/group/learn/databinding/MainActivity.java
package group.learn.databinding;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import androidx.databinding.DataBindingUtil;
import group.learn.databinding.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding activityMainBinding;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        activityMainBinding = DataBindingUtil.setContentView(this,R.layout.activity_main);
        Person person = new Person();
        person.setName("Alfons");
        person.setEmail("example@gmail.com");
        ClickHandler clickHandler = new ClickHandler(this);
        activityMainBinding.setPerson(person);
        activityMainBinding.setClickHandler(clickHandler);
    }
}

File: 00-old-explore/16_custom_list_view/app/src/main/java/com/java/a16_custom_list_view/MainActivity.kt
package com.java.a16_custom_list_view

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ListView
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //1 Adapter a ListView
        val listView : ListView = findViewById(R.id.listworldcupcontainer)

        //2 Data Source
        val dataModels = ArrayList<CountryModelClass>()
        dataModels.add(CountryModelClass("Germany","3",R.drawable.ic_launcher_foreground))
        dataModels.add(CountryModelClass("Indonesia","5",R.drawable.ic_launcher_background))

        //3 Adapter
        val adapter =  CustomAdapter(
            applicationContext,
            dataModels,
        )
        listView.adapter  = adapter

        //on click listener

        listView.setOnItemClickListener { parent, view, position, id ->
            Toast.makeText(
                this@MainActivity,
                "Country ${adapter.getItem(position)?.countryName}",
                Toast.LENGTH_LONG
                ).show()
        }
    }
}